### PR TITLE
Add stories, frontend coverage, fix mui random classnames in snapshots

### PR DESCRIPTION
**Pull request by illume**
# Some storybook improvements

- Stop Material UI from using using random classNames in storyshots (so do not have to regenerate other unrelated snapshots when randomly assigned class names change)
- Setup coverage testing, run on CI
- Add Resource/ViewButton.stories
- Reorganize Resource components together (to find related components)
- Rename Tab to Tabs (file name should be component name)

# How to use

```
cd frontend
CI=true npm run test -- --coverage
npm run storybook
```

### CODE REVIEW

File: frontend/src/components/common/Tabs.tsx [hunk 1]
```diff
@@ -10,13 +10,13 @@ function a11yProps(index: number) {
   };
 }
 
-export interface Tab {
+export interface TabItem {
```
Comment (illume) [inline frontend/src/components/common/Tabs.tsx:13]: This was renamed because Tab interface was shadowing the imported Tab.

Comment (joaquimrocha) [reply frontend/src/components/common/Tabs.tsx:13]: You could import it as `MuiTab`

File: frontend/src/components/common/Resource/ConfirmButton.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import Button, { ButtonProps } from '@material-ui/core/Button';
 import React from 'react';
-import { ConfirmDialog } from './Dialog';
+import { ConfirmDialog } from '../Dialog';
```
Comment (joaquimrocha) [inline frontend/src/components/common/Resource/ConfirmButton.tsx:3]: I don't think the ConfirmButton has something to do with a cluster resource. What am I missing?

File: frontend/src/components/common/Tabs.tsx (hunk 1 above)
Comment (illume) [reply frontend/src/components/common/Tabs.tsx]: Ah yes, that's better.

File: frontend/src/components/common/Resource/ConfirmButton.tsx (hunk 2 above)
Comment (illume) [reply frontend/src/components/common/Resource/ConfirmButton.tsx]: Yes, my mistake. Moved it back.

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**